test/edu/ucsb/nceas/metacat/dataone/D1AuthHelperTest.java

```diff
@@ -255,6 +255,43 @@ public void testDoAdminAuthorization_notAuthorized() throws Exception {
         authDelMock.doAdminAuthorization(session);
     }
 
+    /**
+     * Confirm that 'doAdminAuthorization' throws NotAuthorized exception with session is not null
+     * and subject is null.
+     */
+    @Test(expected = NotAuthorized.class)
+    public void testDoAdminAuthorization_nullSessionSubject() throws Exception {
+        Session sessionLocalNodeAdmin = new Session();
+        sessionLocalNodeAdmin.setSubject(
+            TypeFactory.buildSubject(null));
+
+
+        authDelMock.doAdminAuthorization(null);
+    }
+
+    /**
+     * Confirm that 'doAdminAuthorization' throws NotAuthorized exception with session is not null
+     * and subject is empty.
+     */
+    @Test(expected = NotAuthorized.class)
+    public void testDoAdminAuthorization_emptySessionSubject() throws Exception {
+        Session sessionLocalNodeAdmin = new Session();
+        sessionLocalNodeAdmin.setSubject(
+            TypeFactory.buildSubject(""));
+
+        authDelMock.doAdminAuthorization(null);
```

Comment:
It seems you still test the scenario that the session is null.

Reply:
I have amended the test to correctly check the relevant session, and renamed the variable to be extra clear.